contacts/views.py

```diff
@@ -1,230 +1,157 @@
-    return HttpResponse("Not delete id issue while try delete")
-
-
-def authenticate_user(request):
-    user_str = str(request.user)
-    if request.user is not None:
-        return HttpResponseRedirect('/user_index/')
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework import permissions, status
+from rest_framework.authtoken.models import Token
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.status import (
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_200_OK
+)
+from rest_framework.response import Response
+from rest_framework.views import APIView
+import datetime
+from contacts.serializers import UserSerializer, ContactSerializer, EmailSerializer, UserDetailSerializer, \
+    ContactDetailSerializer
+from contacts.models import User, Contact, Email
+from django.contrib.auth import authenticate
+
+
+class UserProfile(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        if 'token' in request.COOKIES:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            contacts = self.get_serialized(pk, user)
+            serializer = UserSerializer(user)
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("Your token expire Login again", status=HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        if 'token' in request.COOKIES:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            payload = request.data
+            serializer = UserSerializer(user, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("PUT REQUEST FAILED", status=HTTP_400_BAD_REQUEST)
+
+    def get_serialized(self, pk, user):
+
+        queryset = Contact.objects.filter(user_id=pk)
+
+        serializer = ContactSerializer(queryset, many=True)
+        data = serializer.data
+        return data
+
+
+class UserContacts(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        pk = request.COOKIES.get('user_id')
+        _id = int(pk)
+        user = User.objects.get(id=_id)
+        contact = Contact.objects.filter(user_id=user.id)
+        serializer = ContactSerializer(contact, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = ContactSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors)
+
+    def put(self, request):
+        if 'token' in request.COOKIES:
```

Comment:
Use proper authentication and authorization with the token.

Reply:
done